perfkitbenchmarker/configs/option_decoders.py

```diff
@@ -187,33 +187,38 @@ class FloatDecoder(TypeVerifier):
     min: None or float. If provided, it specifies the minimum accepted value.
   """
 
-  def __init__(self, component, option, max=None, min=None, **kwargs):
-    super(FloatDecoder, self).__init__(component, option, (float,), **kwargs)
+  def __init__(self, option, max=None, min=None, **kwargs):
+    super(FloatDecoder, self).__init__(option, (float, int), **kwargs)
     self.max = max
     self.min = min
 
-  def Decode(self, value):
-    """Verifies that the provided value is an int.
+  def Decode(self, value, component_full_name, flag_values):
+    """Verifies that the provided value is a float.
 
     Args:
       value: The value specified in the config.
+      component_full_name: string. Fully qualified name of the configurable
+          component containing the config option.
+      flag_values: flags.FlagValues. Runtime flag values to be propagated to
+          BaseSpec constructors.
 
     Returns:
-      int. The valid value.
+      float. The valid value.
 
     Raises:
       errors.Config.InvalidValue upon invalid input value.
     """
-    value = super(FloatDecoder, self).Decode(value)
+    value = super(FloatDecoder, self).Decode(value, component_full_name,
+                                             flag_values)
     if value is not None:
       if self.max and value > self.max:
```

Comment:
This should probably be ```if self.max is not None and...``` to handle 0 properly (self.min and IntDecoder have this problem too). I can fix this and add tests as a later pull request.